locksmith/src/token_middleware.js

```diff
@@ -0,0 +1,109 @@
+var Web3 = require('web3')
+var isEqual = require('lodash.isequal')
+
+const extractToken = req => {
+  if (
+    req.headers.authorization &&
+    req.headers.authorization.split(' ')[0] === 'Bearer'
+  ) {
+    return req.headers.authorization.split(' ')[1]
+  } else {
+    return null
+  }
+}
+
+var base64Decode = data => {
+  const buff = Buffer.from(data, 'base64')
+  return buff.toString('utf-8')
+}
+
+const validateHaders = headers => {
```

Comment:
```suggestion
const validateHeaders = headers => {
```

Typo, need to correct validateHaders everywhere. validateHaters works if you don't like validateHeaders ;)